ui/packages/catalog-ui-search/src/main/webapp/js/model/MetacardAction.js

```diff
@@ -11,17 +11,32 @@
  **/
 var Backbone = require('backbone');
 require('backbone-associations');
+const URL = require('url');
+const QS = require('querystring');
 
 module.exports = Backbone.AssociatedModel.extend({
     defaults: function() {
         return {
             url: undefined,
             title: undefined,
             description: undefined,
-            id: undefined
+            id: undefined,
+            queryId: undefined
         };
     },
     getExportType: function() {
         return this.get('title').replace(/^Export( as)?\s+\b/, '');
+    },
+    initialize: function() {
+        this.handleQueryId();
+        this.listenTo(this, 'change:queryId', this.handleQueryId);
+    },
+    handleQueryId: function() {
+        if (this.get('queryId') !== undefined) {
+            const parsedU = URL.parse(this.get('url'));
+            const parsedQs = QS.parse(parsedU.query);
+            parsedQs.queryId = this.get('queryId');
+            this.set('url', `${parsedU.protocol}//${parsedU.host}${parsedU.pathname}?${QS.stringify(parsedQs)}`);
```

Comment:
We might need to create the url string without the template literal, looks like they aren't supported in IE11 https://caniuse.com/#feat=template-literals. I would double check with @andrewkfiedler and @djblue though.